ntrack/listener.go

```diff
@@ -0,0 +1,92 @@
+package ntrack
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"sync/atomic"
+
+	"github.com/pkg/errors"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+type trackingListener struct {
+	net.Listener
+	stats *Stats
+}
+
+func NewInstrumentedListener(lis net.Listener) (net.Listener, *Stats) {
```

Comment:
Instead of using context.TODO() what about passing a ctx on the contructor and hold to it? This way caller can even add their own tags and modiiy the views accordingly.